aten/src/ATen/native/LossCTC.cpp

```diff
@@ -155,6 +164,12 @@ std::tuple<Tensor, Tensor> ctc_loss_cpu_template(const Tensor& log_probs, const
       auto log_alpha_a = log_alpha_a_global[b];
       int64_t tg_batch_offset = tg_batch_offsets[b];
 
+      if (input_length == 0) {
```

Comment:
At a high level, this makes sense, but could you give more detail on the existing bug for 0 length cases/which issue in the description this part addresses please?

Reply:
Thanks for your review!

The test case `test_CTCLoss_zero_lengths` gets wrong result and wrong gradient on torch==2.2.2+cu121.

Besides, by modifying lengths to zeros (because all the issues use negative lengths), we can see
1. On CPU, Issues #88047 and #89208 demostrate that the CPU loss_ctc gets wrong result.
2. On CUDA, Issues #84827 and #86596 demostrate that the CPU loss_ctc gets wrong result.